lumen/ai/actor.py

```diff
@@ -172,6 +175,32 @@ def get_prompt_template(cls, key: str = "main") -> str:
         """
         return cls._lookup_prompt_key(cls, key, "template").read_text()
 
+    @property
+    def hash(self) -> str:
```

Comment:
Wonder if it wouldn't be more consistent to implement `to_spec` methods and then hash the output of those.